### ISSUE TITLE
Help - Android Q  apps can no longer execute binaries in their home directory 

### ISSUE THREAD

**Issue by JohnWang1023** [CypherpunkArmory/UserLAnd#827]
# Describe the issue you are having.
https://developer.android.com/preview/behavior-changes-all#execute-permission
https://issuetracker.google.com/issues/128554619
Android Q block exec() on application data files for targetAPI >= Q, will this break Userland?

# Describe how your task behaves in a regular Linux environment
N/A
# Describe how the task you are trying to accomplish behaves in a regular Linux environment.
N/A

**Comment by corbinlc** [comment]
UserLAnd will continue working for a couple of reasons:
1) It only effects your app if you make Q your target API.  We can choose to postpone this if needed for quite a while, but will probably move forward to Q fairly early on because...
2) Once we target Q we can move PRoot and a few other core requirements into the lib directory.  Once that is done, everything will be else will be executed through PRoot so it will all work out fine.
So, there is work we need to do, but nothing that is going to stop people from enjoying UserLAnd.

**Comment by corbinlc** [comment]
I am closing this now as we are aware of this issue.

FYI... we are already working on supporting Q and have made many changes over the last month that set us up for doing so.  We have a little more to do, but we are not using this issue to track that work.

**Comment by MatthewTighe** [comment]
This issue is actually fixed in the latest release. There is a little bit more unrelated stuff that needs to be done for Q compliance.

**Comment by knavels** [comment]
@MatthewTighe @corbinlc  , I get userland installed on my xiaoni ni 9 phone with android 10 and wanted to run code-server and got the message that cannot execute binary file
is this thread related to the error I've got? and is there any fix for this?
